twinkle.js

```diff
@@ -258,33 +267,38 @@ Twinkle.addPortlet = function(navigation, id, text, type, nextnodeid) {
 
 	// verify/normalize input
 	var skin = mw.config.get('skin');
-	type = skin === 'vector' && type === 'menu' && (navigation === 'left-navigation' || navigation === 'right-navigation') ? 'menu' : '';
-	var outerDivClass;
-	var innerDivClass;
+	if (skin !== 'vector' || (navigation !== 'left-navigation' && navigation !== 'right-navigation')) {
+		type = ''; // menu supported only in vector's #left-navigation & #right-navigation
+	}
+	var outerDivClass, innerDivClass;
 	switch (skin) {
 		case 'vector':
+			// XXX: portal doesn't work
 			if (navigation !== 'portal' && navigation !== 'left-navigation' && navigation !== 'right-navigation') {
 				navigation = 'mw-panel';
 			}
 			outerDivClass = navigation === 'mw-panel' ? 'portal' : type === 'menu' ? 'vectorMenu' : 'vectorTabs';
-			innerDivClass = navigation === 'mw-panel' ? 'body' : type === 'menu' ? 'menu' : '';
 			break;
 		case 'modern':
 			if (navigation !== 'mw_portlets' && navigation !== 'mw_contentwrapper') {
-				navigation = 'mw_portlets';
+				navigation = 'p-cactions';
```

Comment:
Should've clarified this before. This change was a result of copy-pasting code from #721 that I didn't refactor. 

Note that even though we are inside `case 'modern'` nothing of this ever gets executed in modern skin - because [`Twinkle.addPortlet` is not executed at all when portletArea is null](https://github.com/azatoth/twinkle/blob/d7e1db398eba8e1814325d2ff885f560552d4925/twinkle.js#L370), that being the case [for modern skin](https://github.com/azatoth/twinkle/blob/d7e1db398eba8e1814325d2ff885f560552d4925/twinkle.js#L144).

Think of this as just prepares for solving #716, for which making this `p-cactions` is needed.

Reply:
Okay thanks, that makes a lot of sense.  Sorry if my schedule/lack thereof has made your life difficult on all these changes these last few months.

I know you hate this, but any chance of undoing that change here?  Would be cleaner when viewing diffs/history, and if it's just for the eventual work in #716, it'd be easier when reviewing that.

Reply:
yeah :) leaving this here doesn't make sense. I guess I was just worried that I'd forget it if/when i work on #716 (and hence have to study the portlet system all over again). 

> Sorry if my schedule/lack thereof has made your life difficult on all these changes these last few months.

oh no! not at all. Frankly, I just love writing these PRs. I'm hardly bothered even if no one reviews them for a loong while.

Reply:
🤗